purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/PaywallTransition.kt

```diff
@@ -0,0 +1,91 @@
+package com.revenuecat.purchases.paywalls.components
+
+import com.revenuecat.purchases.InternalRevenueCatAPI
+import com.revenuecat.purchases.utils.serializers.EnumDeserializerWithDefault
+import com.revenuecat.purchases.utils.serializers.SealedDeserializerWithDefault
+import dev.drewhamilton.poko.Poko
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Defines how a paywall screen is transitioned when it initially appears.
+ *
+ * @property type The type of transition to use. Defaults to [TransitionType.Fade].
+ * @property displacementStrategy Determines how/when the view hierarchy is displaced by the view being animated in.
+ * @property animation Additional animation configuration for the transition.
+ */
+@InternalRevenueCatAPI
+@Poko
+@Serializable
+class PaywallTransition(
+    @get:JvmSynthetic val type: TransitionType = TransitionType.Fade,
+    @get:JvmSynthetic
+    @SerialName("displacement_strategy")
+    val displacementStrategy: DisplacementStrategy,
+    @get:JvmSynthetic val animation: PaywallAnimation? = null,
+) {
+
+    /**
+     * Determines how the view being animated out is displaced by the view being animated in.
+     *
+     * A [GREEDY] displacement will result in the space being taken up by the incoming view
+     * *before* it attempts to transition into the view hierarchy.
+     *
+     * A [LAZY] displacement will not do this, instead it will result in shifting the layout
+     * as the new view inserts itself.
+     */
+    @InternalRevenueCatAPI
+    @Serializable(with = DisplacementStrategyDeserializer::class)
+    enum class DisplacementStrategy {
+        @SerialName("greedy")
+        GREEDY,
+
+        @SerialName("lazy")
+        LAZY,
+    }
+
+    /**
+     * Defines the type of transition to use for paywall transitions.
+     *
+     * [NOTE] This is a sealed class and not an enum because we see a future where we may want
+     * to pass back more verbose instructions to the view layer than a simple enum case
+     */
+    @InternalRevenueCatAPI
+    @Serializable(with = TransitionTypeDeserializer::class)
+    sealed class TransitionType {
```

Comment:
Now that we've removed the `Custom`, we could make this an enum, same for `AnimationType`. Since it's annotated with `InternalRevenueCatAPI`, we can always change it later if needed.